newIDE/app/src/InAppTutorial/BlockingLayerWithHoles.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+import * as React from 'react';
+import { aboveMaterialUiMaxZIndex } from '../UI/MaterialUISpecificUtil';
+import { useInterval } from '../Utils/UseInterval';
+
+const blockingLayerZIndex = aboveMaterialUiMaxZIndex;
+export const itemAboveBlockingLayerZIndex = blockingLayerZIndex + 1;
+
+type Props = {|
+  elements: Array<HTMLElement>,
+|};
+
+const BlockingLayerWithHoles = ({ elements }: Props) => {
+  const [holes, setHoles] = React.useState([]);
+
+  const updateHoles = React.useCallback(
+    () => {
+      const newHoles = elements.map(element => {
+        const { top, left, width, height } = element.getBoundingClientRect();
+        return {
+          top,
+          left,
+          width,
+          height,
+        };
+      });
+      setHoles(newHoles);
+    },
+    [elements]
+  );
+
+  React.useEffect(
+    () => {
+      // Update hole position on scroll & resize
+      window.addEventListener('wheel', updateHoles);
+      window.addEventListener('touchmove', updateHoles);
+      window.addEventListener('resize', updateHoles);
+      updateHoles(); // Initial position update
+
+      return () => {
+        window.removeEventListener('wheel', updateHoles);
+        window.removeEventListener('touchmove', updateHoles);
+        window.removeEventListener('resize', updateHoles);
+      };
+    },
+    [updateHoles]
+  );
+
+  // We also update the hole position every second, as the item may keep moving
+  // on the deceleration phase of a scroll.
+  useInterval(updateHoles, 1000);
```

Comment:
```suggestion
  useInterval(updateHoles, elements.length === 0 ? null : 1000);
```
Or something like that?

Reply:
I'm updating the logic as mentioned in my previous comment